Simplenote/src/main/java/com/automattic/simplenote/TagDialogFragment.java

```diff
@@ -81,80 +103,60 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
         mDialogEditTag = builder.create();
         mDialogEditTag.setOnShowListener(this);
 
-        mClickListenerNegativeRename = new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                dismiss();
-            }
-        };
-
-        mClickListenerPositiveRename = new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                String tagNew = getTagNew();
-
-                if (tagNew.equals(mTagOld)) {
-                    dismiss();
-                }
+        mClickListenerNegativeRename = v -> viewModel.close();
+        mClickListenerPositiveRename = v -> viewModel.renameTagIfValid();
+        mClickListenerNeutralConflict = v -> showDialogRenameTag();
 
-                int index = mTag.hasIndex() ? mTag.getIndex() : mBucketTag.count();
-                boolean isRenamingToLexicalTag = TagUtils.hashTag(tagNew).equals(TagUtils.hashTag(mTagOld));
-                boolean hasCanonicalTag = TagUtils.hasCanonicalOfLexical(mBucketTag, tagNew);
+        mClickListenerPositiveConflict = v -> viewModel.renameTag();
+    }
 
-                if (hasCanonicalTag && !isRenamingToLexicalTag) {
-                    String tagCanonical = TagUtils.getCanonicalFromLexical(mBucketTag, tagNew);
-                    showDialogErrorConflict(tagCanonical, mTagOld);
-                    return;
-                }
 
-                tryToRenameTag(tagNew, index);
+    private void setObservers() {
+        viewModel.getUiState().observe(this, uiState -> {
+            // Validate if the current state has an error
+            if (uiState.getErrorMsg() != null) {
+                mEditTextLayout.setError(getString(uiState.getErrorMsg()));
+                mButtonPositive.setEnabled(false);
+            } else {
+                // If there is not an error, enable save button
+                mEditTextLayout.setError(null);
+                mButtonPositive.setEnabled(true);
             }
-        };
+        });
 
-        mClickListenerNeutralConflict = new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                showDialogRenameTag();
-            }
-        };
-
-        mClickListenerPositiveConflict = new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                String tagNew = getTagNew();
-                int index = mTag.hasIndex() ? mTag.getIndex() : mBucketTag.count();
-                tryToRenameTag(tagNew, index);
+        viewModel.getEvent().observe(this, event -> {
+            if (event instanceof CloseEvent || event instanceof FinishEvent) {
+                dismiss();
+            } else if (event instanceof ShowErrorEvent) {
+                Context context = requireContext();
+                DialogUtils.showDialogWithEmail(
```

Comment:
Minor (🔍): Consider making this a one-liner.

Reply:
Fixed in b112ade6ed25423db196d9bdf6f5c0b33f60979d.